aws/amazon-chime-sdk-component-library-react
# Avoid generating new audio input devices using deep comparison

**Issue #:** N/A

**Description of changes:**
`useAudioInputs` keep generating a new instance of the devices array, so we can not have devices as a dependency in the VF component, instead, we use devices.length and devices[0].label as the dependencies which is annoying. Implement deep comparison through `lodash` function to avoid this issue.

**Testing**
1. Have you successfully run `npm run build:release` locally? Yes

2. How did you test these changes?
* Join the meeting in Chrome 
* Keep the default selected device as audio input 
* Enable Amazon Voice Focus 
* Apply a new audio device (AirPods) in the middle of meeting 
* Current audio device switches to AirPods voice focus transform device in the dropdown list.
* You should see a check against this newly connected device now.

3. If you made changes to the component library, have you provided corresponding documentation changes? N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
